apps/meteor/ee/app/livechat-enterprise/server/api/lib/priorities.ts

```diff
@@ -46,6 +39,78 @@ export async function findPriorities({ text, pagination: { offset, count, sort }
 	};
 }
 
-export async function findPriorityById({ priorityId }: FindPrioritiesByIdParams): Promise<FindPrioritiesByIdResult> {
-	return LivechatPriority.findOneById(priorityId);
+export async function updatePriority(
+	_id: string,
+	data: Pick<ILivechatPriority, 'name'> & { reset?: boolean },
+): Promise<ILivechatPriority | null> {
+	const query = {
+		_id,
+	};
+	const update: Pick<UpdateFilter<ILivechatPriority>, '$set' | '$unset'> = {
+		...((data.reset && {
+			$set: { dirty: false },
+			$unset: { name: 1 },
+		}) || {
+			// Trim value before inserting
+			$set: { name: data.name?.trim(), dirty: true },
+		}),
+	};
+
+	if (data.name) {
+		// If we want to enforce translated duplicates we need to change this
+		const priority = await LivechatPriority.findOne({ name: new RegExp(`^${escapeRegExp(data.name.trim())}$`, 'i') });
+		if (priority && priority._id !== _id) {
+			throw new Error('error-duplicate-priority-name');
+		}
+	}
+
+	const createdResult = await LivechatPriority.findOneAndUpdate(query, update, {
+		returnDocument: 'after',
+	});
+
+	if (!createdResult.ok || !createdResult.value) {
+		logger.error(`Error updating priority: ${_id}. Unsuccessful result from mongodb. Result`, createdResult);
+		throw Error('error-unable-to-update-priority');
+	}
+
+	return createdResult.value;
 }
+
+export const updateRoomPriority = async (rid: string, user: IUser, priorityId: string): Promise<void> => {
+	const room = await LivechatRooms.findOneById(rid, { projection: { _id: 1 } });
+	if (!room) {
+		throw new Error('error-room-does-not-exist');
+	}
+
+	const priority = await LivechatPriority.findOneById(priorityId);
```

Comment:
if you just want to check if a record exists, pls use a `projection` to fetch only the `_id` field, just like on the line above.

Reply:
In this case, we're not just checking if record exists, but also passing this variable to further where other properties within it will be used. Hence we don't need to apply projection in this case.